mindmeld/auto_annotator.py

```diff
@@ -835,5 +901,320 @@ def valid_entity_check(self, entity):
         return Entity.is_system_entity(entity) or entity in self.supported_entity_types
 
 
-register_annotator("SpacyAnnotator", SpacyAnnotator)
-register_annotator("BootstrapAnnotator", BootstrapAnnotator)
+class NoTranslationDucklingAnnotator(Annotator):
+    """Custom Annotator class used to generate annotations."""
+
+    def __init__(self, *args, **kwargs):
+        """Initializes a NoTranslationDucklingAnnotator.
+
+        Args:
+            app_path (str): The location of the MindMeld app.
+            annotation_rules (list): List of Annotation rules.
+            language (str, optional): Language as specified using a 639-1/2 code.
+            locale (str, optional): The locale representing the ISO 639-1 language code and \
+                ISO3166 alpha 2 country code separated by an underscore character.
+            overwrite (bool): Whether to overwrite existing annotations with conflicting spans.
+            unannotate_supported_entities_only (bool): Only allow removal of supported entities.
+            unannotation_rules (list): List of Annotation rules.
+        """
+        super().__init__(*args, **kwargs)
+
+    def parse(self, sentence, entity_types=None, **kwargs):
+        """
+        Args:
+            sentence (str): Sentence to detect entities.
+            entity_types (list): List of entity types to parse. If None, all
+                    possible entity types will be parsed.
+        Returns: entities (list): List of entity dictionaries.
+        """
+        duckling_candidates = self.duckling.get_candidates_for_text(
+            sentence,
+            entity_types=entity_types,
+            language=self.language,
+            locale=self.locale,
+        )
+        filtered_candidates = (
+            NoTranslationDucklingAnnotator._filter_out_bad_duckling_candidates(
+                duckling_candidates
+            )
+        )
+        spans = [
+            Span(candidate["start"], candidate["end"] - 1)
+            for candidate in filtered_candidates
+        ]
+        final_spans = (
+            NoTranslationDucklingAnnotator._get_largest_non_overlapping_candidates(
+                spans
+            )
+        )
+        final_candidates = []
+        for span in final_spans:
+            for candidate in filtered_candidates:
+                if span == Span(candidate["start"], candidate["end"] - 1):
+                    final_candidates.append(candidate)
+                    break
+        if entity_types:
+            final_candidates = [
+                e for e in final_candidates if e["entity_type"] in entity_types
+            ]
+        return [
+            duckling_item_to_entity_dict(candidate) for candidate in final_candidates
```

Comment:
return a list of entity objects as discussed

Reply:
All annotators now return QueryEntity objects. Tests have been updated accordingly, working on updating the docs to reflect this change as well.